### PR TITLE
Run transcription in the cloud

### PR DESCRIPTION
**Pull request by isinyaaa**
Use [Replicate](https://replicate.com) to run transcriptions. Because they have limited support for direct uploads, also use AWS S3 to serve files.

Depends on: #97

### CODE REVIEW

File: superlesson/steps/transcribe.py [hunk 1]
```diff
@@ -1,113 +1,267 @@
...
         bench_start = datetime.now()
 
-        if self._has_nvidia_gpu():
-            model = WhisperModel(model_size, device="cuda", compute_type="int8_float16")
-        else:
-            model = WhisperModel(model_size, device="cpu", cpu_threads=cpu_count(), compute_type="auto")
+        if not local and getenv("REPLICATE_API_TOKEN"):
+            with open(audio_path, "rb") as file:
+                data = file.read()
+                s3_name = sha256(data).hexdigest()
```
Comment (isinyaaa) [inline superlesson/steps/transcribe.py:47]: Does it make sense to have a name that's easier to identify?
e.g. `compilers-5-verylonghashhere123456789`

File: superlesson/steps/transcribe.py [hunk 2]
```diff
@@ -1,113 +1,267 @@
...
+            s3.upload_fileobj(f, cls._bucket_name, name)
+
+        logging.info(f"{file} uploaded to S3 as {name}")
+        return f"https://{cls._bucket_name}.s3.amazonaws.com/{name}"
+
+    @classmethod
+    def _local_transcription(cls, transcription_path: Path, model_size: str):
+        from os import cpu_count
+
+        from faster_whisper import WhisperModel
```
Comment (isinyaaa) [inline superlesson/steps/transcribe.py:114]: I couldn't find a package for whisperX, no idea how we'd use that locally, but it's actually possible using docker.

File: superlesson/steps/transcribe.py [hunk 3]
```diff
@@ -1,113 +1,267 @@
...
+class Segment:
+    text: str
+    start: float
+    end: float
+
+
 class Transcribe:
     """Class to transcribe a lesson."""
 
+    _bucket_name = "lesson-audios"
```
Comment (isinyaaa) [inline superlesson/steps/transcribe.py:23]: should this be configurable by the user?

File: README.md [hunk 4]
```diff
@@ -6,18 +6,30 @@
 
 ### Setup
 
-SuperLesson (SL) uses
-[faster-whisper](https://github.com/guillaumekln/faster-whisper) in order to
-transcribe audio. You can use your GPU to run it faster, but you need to install
-some dependencies. Refer to their documentation for more details.
+SuperLesson (SL) uses [WhisperX](hhttps://github.com/m-bain/whisperX) through [Replicate](https://replicate.com) to transcribe audio.
```
Comment (isinyaaa) [inline README.md:9]: Should we add https://replicatestatus.com/ here for reference?

**Review by rsgrava — APPROVED** [review]

**Pull request merged**